fractions/keycloak-microprofile-jwt/src/main/java/org/wildfly/swarm/keycloak/mpjwt/deployment/KeycloakJWTCallerPrincipalFactory.java

```diff
@@ -12,21 +13,37 @@
 import io.smallrye.jwt.auth.principal.JWTCallerPrincipal;
 import io.smallrye.jwt.auth.principal.JWTCallerPrincipalFactory;
 import io.smallrye.jwt.auth.principal.ParseException;
+
 /**
  * An implementation of the abstract JWTCallerPrincipalFactory that uses the Keycloak token parsing classes.
  */
 public class KeycloakJWTCallerPrincipalFactory extends JWTCallerPrincipalFactory {
+    private static final Logger log = Logger.getLogger(KeycloakJWTCallerPrincipalFactory.class);
+
+    private static final KeycloakDeployment deployment = createDeployment();
 
     @Override
     public JWTCallerPrincipal parse(final String token, final JWTAuthContextInfo authContextInfo) throws ParseException {
         try {
-            InputStream keycloakJson = Thread.currentThread().getContextClassLoader().getResourceAsStream("keycloak.json");
-            KeycloakDeployment deployment = KeycloakDeploymentBuilder.build(keycloakJson);
             JWSInput jwsInput = new JWSInput(token);
             AccessToken accessToken = AdapterTokenVerifier.verifyToken(jwsInput.getWireString(), deployment);
             return new KeycloakJWTCallerPrincipal(jwsInput.readContentAsString(), accessToken);
         } catch (Throwable ex) {
             throw new ParseException("Failure to parse the token", ex);
         }
     }
+
+    private static KeycloakDeployment createDeployment() {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        InputStream keycloakJson = cl.getResourceAsStream("keycloak.json");
+        if (keycloakJson == null) {
+            keycloakJson = cl.getResourceAsStream("WEB-INF/keycloak.json");
+        }
+        if (keycloakJson == null) {
+            log.warn("keycloak.json resource is not available");
+            return null;
```

Comment:
Also the `null` return value here will just cause a NPE later, no?

Reply:
Right, there will be a warning, but no NPE visible (will be caught by the Throwable catch block) - the user will see JWT authentication failed or something like that, this is what microprofile-jwt will log indirectly. Not sure throwing something at the static initialization time is worth it, may be some more work can be done later on, checking the YAML properties, etc. Do you prefer IllegalStateException (or similar) at this stage ?

Reply:
My response here got lost too... NPE won't be leaking, the users will see microprofile-jwt logging 'JWT authentication failed' or something like that.  Perhaps, the code creating the KC principal can be enhanced later on to check the system properties (originally set as YAML properties) if the deployment is null...

Reply:
Ah true, I didn't notice the `catch` block. This is probably fine for now :-)